crates/bevy_core_pipeline/src/taa/mod.rs

```diff
@@ -364,51 +356,53 @@ fn extract_taa_settings(mut commands: Commands, mut main_world: ResMut<MainWorld
         let mut entity_commands = commands
             .get_entity(entity)
             .expect("Camera entity wasn't synced.");
-        if camera.is_active && has_perspective_projection {
-            entity_commands.insert(taa_settings.clone());
-            taa_settings.reset = false;
+        if taa_settings.is_some() && camera.is_active && has_perspective_projection {
```

Comment:
I bet this same bug is lurking a few other places.

Reply:
I looked at how ExtractComponentPlugin works and I think all of those will have the same issue too, since it's doing `type QueryData = &'static Self;`. Right? Double check me on this please.

Reply:
Yes, that's immediately what came to mind since it only runs if the component is actually present and just removes components if it doesn't get extracted (e.g. because a camera is disabled). I don't love the idea of essentially running all extracts unconditionally all the time (especially if a component like taa has never even been added) but that's definitely the fix given the current patterns.

Reply:
Yes, I agree, some kind of observer/event based system would be great for this kind of data that can change but doesn't need to be extracted every frame.